proto/dataplane.proto

```diff
@@ -29,72 +29,101 @@ enum Error {
   ERROR_UNKNOWN_ERROR = 3;
 }
 
-enum IfType {
-  IF_TYPE_UPLINK = 0;
-  IF_TYPE_MANAGEMENT = 1;
-  IF_TYPE_VXLAN = 2;
-}
-
 message GetConfigGenerationRequest {
 }
 
 message GetConfigGenerationResponse {
   uint64 generation = 1;
 }
 
-message Subnet {
-  string cidr = 1;
-  string name = 2;
+
+/* ================================================ */
+/* ==== Interfaces (both underlat and overlay) ==== */
+/* ================================================ */
+
+message Interface {
+  string name = 1;
+  string ipaddr = 2;
+  IfType type = 3;
+  IfRole role = 4;
 }
 
-message VPC {
-  string id = 1;
-  string name = 2;
-  uint32 vni = 3;
-  repeated Subnet subnets = 4;
+enum IfType {
+  IF_TYPE_ETHERNET = 0;
+  IF_TYPE_VLAN = 1;
+  IF_TYPE_LOOPBACK = 2;
+  IF_TYPE_VTEP = 3;
 }
 
-message Device {
-  uint32 index = 1;
-  string ipaddr = 2;
-  string name = 3;
-  string pciaddr = 4;
-  IfType type = 5;
+enum IfRole {
+  IF_ROLE_FABRIC = 0;
+  IF_ROLE_MANAGEMENT = 1;
+  IF_ROLE_EXTERNAL = 2;
 }
 
-message PeeringIPs {
+/* ================= */
+/* ==== Overlay ==== */
+/* ================= */
+
+message VpcSubnetAs {
   string cidr = 1;
-  string not = 2;
+  repeated string exclude = 2;
 }
 
-message PeeringAs {
-  string cidr = 1;
-  string not = 2;
+message VpcSubnet {
+  string name = 1; /* informational subnet name -- may be used for stats */
+  string cidr = 2;
+  repeated string exclude = 3;
+  optional VpcSubnetAs as = 4;
 }
 
-message PeeringEntry {
-  repeated PeeringAs as = 1;
-  repeated PeeringIPs ips = 2;
+message PeeringEntryFor {
+  string vpc = 1;
+  repeated VpcSubnet subnets = 2;
 }
 
-message Peering {
-  string name = 1;
-  map<string, PeeringEntry> entries = 2;
+message VpcPeering {
+  string name = 1; /* unique: e.g. VPC-1--VPC-2 */
+  repeated PeeringEntryFor = 2;
+}
+
+message VPC {
+  string name = 1; /* unique (key) */
+  uint32 vni = 2;
+  repeated Interface interfaces = 3;
```

Comment:
So, does it mean we don't need to pass VPC subnets and information about them?

Reply:
I don't think we need. We learn those via BGP. That's one of the reasons for the changes.
Even if we did not learn them via BGP, the peering objects should contain that information, right ?

Reply:
The peering object only contains the VPC names, not their subnets/etc

Reply:
> The peering object only contains the VPC names and peering rules, not the list of VPC subnets/etc
I believe that the gateway may only need the "ips" and "as" in the peering objects.

Reply:
@Frostman @sergeymatov Let's not merge any of this until we reach an agreement.